src/api.ts

```diff
@@ -35,6 +37,32 @@ export const bas = {
         return ActionsController.getAction(actionId);
     },
 
+    async getParameter(parameterName: string) : Promise<string | null> {
+        const onBasEnv = "env.WS_BASE_URL";
+        const isInBAS = !_.isEmpty(_.get(process, onBasEnv));
+        if (!isInBAS) {
+            getLogger().trace("Running in VS Code, so returning null.");
+            return null;
+        }
+        getLogger().trace("Running on BAS.");
+
+        let sap;
+        try {
+            sap = require('@sap/plugin');
+        } catch (error) {
+            getLogger().error("Failed to load @sap/plugin, so returning null.", {error});
+            return null;
+        }
+        getLogger().trace("@sap/plugin successfully loaded.");
+
+        const configuration = await sap.window.configuration();
+        getLogger().trace("Configuration successfully received.", {configuration});
+
+        const parameterValue = _.get(configuration, parameterName, null);
```

Comment:
should we return `null` or `undefined` ?
you ay be able to use optional chaining instead of `_.get`

Reply:
Works perfect for undefined "const parameterValue = configuration?.[parameterName];", but not for null. Or you know a better way?